ozone/core/management/commands/import_exemptions.py

```
@@ -0,0 +1,651 @@
+
+        if not period.is_reporting_allowed:
+            logger.error(
+                "Error `Reporting cannot be performed for this reporting period.`"
+                "while saving: %s/%s ",
+                party_abbr,
+                period_name,
+                exc_info=True
+            )
+            return False
+
+        if purge:
+            self.delete_instance(party, period, obligation)
+            return True
+
+        if recreate:
+            self.delete_instance(party, period, obligation)
+
+        data = self.get_submission_raf_data(party, period, rows)
+
+        submission = Submission.objects.create(
+            **data["submission"]
+        )
+
+        for key, value in data["submission_info"].items():
+            setattr(submission.info, key, value)
+        submission.info.save()
+
+        for raf in data['rafreports']:
+            # Create RAFReport objects
+            report = RAFReport.objects.create(
+                submission=submission,
+                **raf['raf_report']
+            )
+
+            # Create RAFImport objects, if exists.
+            for raf_import in raf['imports']:
+                RAFImport.objects.create(
+                    report=report,
+                    **raf_import
+                )
+
+        submission._current_state = "finalized"
+        submission.save()
+
+        for obj in submission.history.all():
+            obj.history_user = self.admin
+            obj.save()
+
+        return True
+
+    def proccess_submission_exemption_entry(self, party_abbr, period_name, rows, obligation, recreate=False, purge=False):
+        """
+        Inserts the processed data into the DB.
+        """
+
+        try:
+            party = self.parties[party_abbr]
+            period = self.periods[period_name]
+        except KeyError as e:
+            logger.critical("Unable to find matching %s: %s", e, rows)
+            return False
+
+        if not period.is_reporting_allowed:
```

Hmmm, I don't think we should raise this error, we are importing historical submissions here after all.

---

It's just a logger that emphasizes the fact that the current period has `
is_reporting_allowed=False`. The problem is that the legacy data contains submissions from 2019.